sky/adaptors/aws.py

```diff
@@ -34,7 +34,31 @@
 boto3 = None
 botocore = None
 _session_creation_lock = threading.RLock()
-_local = threading.local()
+
+version = 1
+
+
+class ThreadLocalLRUCache(threading.local):
+
+    def __init__(self, maxsize=32):
+        super().__init__()
+        self.cache = functools.lru_cache(maxsize=maxsize)
+
+
+def thread_local_lru_cache(maxsize=32):
+    # Create thread-local storage for the LRU cache
+    local_cache = ThreadLocalLRUCache(maxsize)
```

Comment:
I think it should only be created once for each function, as it will only be called once when defining the function, and the function will be wrapped by the return decorator?